modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v10_22_1/ResourceActionUpgradeProcess.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.internal.upgrade.v10_22_1;
+
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.ResourceAction;
+import com.liferay.portal.kernel.model.ResourcePermission;
+import com.liferay.portal.kernel.service.ResourceActionLocalServiceUtil;
+import com.liferay.portal.kernel.service.ResourcePermissionLocalServiceUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Mario Gomes
+ */
+public class ResourceActionUpgradeProcess extends UpgradeProcess {
```

Comment:
Rename it to `ObjectEntryFolderPermissionUpgradeProcess`. Rename the test class too.